### PR TITLE
Orbital:  Remove needless GSF for TPV

### PR DESCRIPTION
**Pull request by javierpedrozaing**
Description
-------------------------
[SER-969](https://spreedly.atlassian.net/browse/SER-969) 
[SER-970](https://spreedly.atlassian.net/browse/SER-970) 
[SER-971](https://spreedly.atlassian.net/browse/SER-971)

This commit remove a couple of needless GSF during a TPV flow we hard-coded the values required for token_tx_type and card_brand instead of sending these as GSF

Unit test
-------------------------
Finished in 1.903239 seconds.

148 tests, 839 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 100% passed

77.76 tests/s, 440.83 assertions/s

Remote test
-------------------------
Finished in 118.453815 seconds.

133 tests, 526 assertions, 13 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 90.2256% passed

1.12 tests/s, 4.44 assertions/s

Rubocop
-------------------------
781 files inspected, no offenses detected

### CODE REVIEW

File: lib/active_merchant/billing/gateways/orbital.rb [hunk 1]
```diff
@@ -565,15 +565,15 @@ def get_address(options)
       end
 
       def add_safetech_token_data(xml, payment_source, options)
-        xml.tag! :CardBrand, options[:card_brand]
-        xml.tag! :AccountNum, payment_source
+        xml.tag! :CardBrand, payment_source.split(';')[1]
```
Comment (Heavyblade) [inline lib/active_merchant/billing/gateways/orbital.rb:568]: 💬 Style:

maybe is just matter of taste on semantics => `payment_source.split(';').first`

File: lib/active_merchant/billing/gateways/orbital.rb [hunk 2]
```diff
@@ -1031,7 +1031,7 @@ def build_new_order_xml(action, money, payment_source, parameters = {})
             add_stored_credentials(xml, parameters)
             add_pymt_brand_program_code(xml, payment_source, three_d_secure)
             add_mastercard_fields(xml, payment_source, parameters, three_d_secure) if mastercard?(payment_source)
-            xml.tag! :TokenTxnType, parameters[:token_txn_type] if parameters[:token_txn_type]
+            xml.tag! :TokenTxnType, parameters[:token_txn_type]
```
Comment (Heavyblade) [inline lib/active_merchant/billing/gateways/orbital.rb:1034]: ❓ Question:

Given how strict is Orbital XSD definition and that this line will always include `TokenTxnType`, is this working properly with `refund` and `credit` transactions, because those methods are currently using the `build_new_order_xml`.

**Review by Heavyblade — COMMENTED** [review]
Looking pretty much 👍  @javierpedrozaing , left a couple of comments and a question.

File: lib/active_merchant/billing/gateways/orbital.rb (hunk 1 above)
Comment (javierpedrozaing) [reply lib/active_merchant/billing/gateways/orbital.rb]: you 're right updated

File: lib/active_merchant/billing/gateways/orbital.rb (hunk 2 above)
Comment (javierpedrozaing) [reply lib/active_merchant/billing/gateways/orbital.rb]: you 're right updated, adding a validation

File: lib/active_merchant/billing/gateways/orbital.rb [hunk 3]
```diff
@@ -901,7 +901,7 @@ def commit(order, message_type, retry_logic = nil, trace_number = nil)
             request.call(remote_url(:secondary))
           end
 
-        authorization = order.include?('<TokenTxnType>GT</TokenTxnType>') ? response[:safetech_token] : authorization_string(response[:tx_ref_num], response[:order_id])
+        authorization = order.include?('<TokenTxnType>GT</TokenTxnType>') ? authorization_string(response[:safetech_token], response[:card_brand]) : authorization_string(response[:tx_ref_num], response[:order_id])
```
Comment (naashton) [inline lib/active_merchant/billing/gateways/orbital.rb:904]: I think we need to rethink this `authorization` string we are building. There is a loose pattern for these things that we follow, which is not well documented, but it helps standardize these strings at the individual gateway level.

TLDR; We don't want to conditionally build the string to be different, depending on the transaction. Regardless of the values returned in the response from the gateway, we want to maintain a consistent structure.

Example:

```
authorization = authorization_string(r[field1], r[field2], r[field3])

# authorization = 'val1;val2;val3'
# authorization = 'val1;;val3'
# authorization = 'val1;;'
```

This way, wherever we split this string, we can reliably retrieve the index of the authorization and _know_ that it should be value we expect.

For this, we should take the existing `authorization_string` call and append the values we would like to include in this string, so that the length of this string (after it is split into the individual values) is always the same.

**Review by naashton — CHANGES_REQUESTED** [review]

File: lib/active_merchant/billing/gateways/orbital.rb [hunk 4]
```diff
@@ -901,7 +902,7 @@ def commit(order, message_type, retry_logic = nil, trace_number = nil)
             request.call(remote_url(:secondary))
           end
 
-        authorization = order.include?('<TokenTxnType>GT</TokenTxnType>') ? response[:safetech_token] : authorization_string(response[:tx_ref_num], response[:order_id])
+        authorization = authorization_string(response[:tx_ref_num], response[:order_id], response[:safetech_token], response[:card_brand])
```
Comment (naashton) [inline lib/active_merchant/billing/gateways/orbital.rb:905]: One final question regarding this line. Should we more safely fetch the values from the hash? My concern is if the gateway does not return one of these keys in the response that we'll fail on an index error.

**Review by naashton — APPROVED** [review]
